kliff/transforms/configuration_transforms/graphs/generate_graph.py

```diff
@@ -0,0 +1,199 @@
+from loguru import logger
+from monty.dev import requires
+
+from kliff.utils import torch_available, torch_geometric_available
+
+if torch_available():
+    import torch
+
+from typing import TYPE_CHECKING
+
+import kliff.transforms.configuration_transforms.graphs.graph_module as graph_module
+
+from ..configuration_transform import ConfigurationTransform
+
+if TYPE_CHECKING:
+    from kliff.dataset import Configuration
+
+if torch_geometric_available():
+    from torch_geometric.data import Data
+
+
+@requires(torch_geometric_available(), "torch_geometric is needed for KLIFFTorchGraph")
+class KLIFFTorchGraph(Data):
+    """
+    A Pytorch Geometric compatible graph representation of a configuration. When loaded
+    into a class:`torch_geometric.data.DataLoader` the graphs of type KLIFFTorchGraph
+    will be automatically collated and batched.
+
+    """
+
+    def __init__(self):
+        super(KLIFFTorchGraph, self).__init__()
+        self.num_nodes = (
+            None  # Simplify sizes and frees up pos key word, coords is cleaner
+        )
+        self.energy = None
+        self.forces = None
+        self.n_layers = None
+        self.coords = None
+        self.images = None
+        self.species = None
+        self.z = None
+        self.contributions = None
+
+    def __inc__(self, key, value, *args, **kwargs):
+        if "index" in key or "face" in key:
+            return self.num_nodes
+        elif "contributions" in key:
+            return 2
+        elif "images" in key:
+            return torch.max(value) + 1
+        else:
+            return 0
+
+    def __cat_dim__(self, key, value, *args, **kwargs):
+        if "index" in key or "face" in key:
+            return 1
+        else:
+            return 0
+
+
+class KLIFFTorchGraphGenerator(ConfigurationTransform):
+    """
+    Generate a graph representation of a configuration. This generator will also save the
+    required parameters for porting the model over to KIM-API using TorchMLModelDriver.
+    The configuration file saved here will generate identical graphs at KIM-API runtime.
+    For porting the graph representation you also need to provide the TorchScript model file name.
+
+    Args:
+        species (list): List of species.
+        cutoff (float): Cutoff distance.
+        n_layers (int): Number of convolution layers.
+        as_torch_geometric_data (bool): If True, the graph will be returned as a Pytorch
+            Geometric Data object.
+        implicit_fingerprint_copying (bool): If True, the fingerprint will be copied to
+            the Configuration object's fingerprint attribute.
+    """
+
+    def __init__(
+        self,
+        species,
+        cutoff,
+        n_layers,
+        as_torch_geometric_data=False,
```

Comment:
I've remove `as_torch_geometric_data` here, and `self._to_py_graph(graph)` in __init__. Instead, changing `self._to_py_graph(graph)` to `self.to_pyg_graph(graph)` and call it explicitly when necessary. 

This should make the class more clear.

Reply:
Done. `to_py_graph` now returns `PyGGraph` class, which is initialized as a dummy class with same attributes if PyG is not available.